WordPress/Classes/Services/NullBlogPropertySanitizer.swift

```diff
@@ -33,7 +33,8 @@ import WordPressShared
 
         store.set(currentBuildVersion(), forKey: Self.lastSanitizationVersionNumber)
 
-        let entityNamesWithRequiredBlogProperties = [
+        // FIXME: Not sure why it's necessary to define [String] as the type. Without it, the NSFetchRequest<NSManagedObject>(entityName: entityName) below does not compile.
```

Comment:
I think that's because there are multiple `entityName` definitions with mismatch APIs. For example, `+[PostCategory entityName]` [returns an IUO in Swift](https://github.com/wordpress-mobile/WordPress-iOS/blob/50cb14f968430dcefd0cb6641a91990fe851ded3/WordPress/Classes/Models/PostCategory.h#L16), and the `entityName` on all `NSManagedObject` [returns non-optional String](https://github.com/wordpress-mobile/WordPress-iOS/blob/50cb14f968430dcefd0cb6641a91990fe851ded3/WordPress/Classes/Utility/CoreData/CoreDataHelper.swift#L14). Changing the access level probably also changed how Swift compiler chooses which API to use.

Reply:
Thanks for the deep dive @crazytonyli !